pusher/pusher_client.py

```diff
@@ -41,12 +41,26 @@ def __init__(
             timeout=5, cluster=None,
             encryption_master_key=None,
             json_encoder=None, json_decoder=None,
-            backend=None, **backend_options):
+            backend=None, max_num_channels = None,
+            max_len_event_name = None, max_len_data = None, **backend_options):
         super(PusherClient, self).__init__(
             app_id, key, secret, ssl, host, port, timeout, cluster,
             encryption_master_key, json_encoder, json_decoder,
             backend, **backend_options)
 
+        self.max_num_channels = 100
```

Comment:
It might be better to do something like:

```suggestion
        self.max_num_channels = max_num_channels if max_num_channels is not None else 100
```

Reply:
Similar changes can be applied below and the following `if` statements can be removed.